process.md

```diff
@@ -4,7 +4,19 @@ Swift is a powerful and intuitive programming language that is designed to make
 
 ## Scope
 
-The Swift evolution process covers all changes to the Swift language and the public interface of the Swift standard library, including new language features and APIs (no matter how small), changes to existing language features or APIs, removal of existing features, and so on. Smaller changes, such as bug fixes, optimizations, or diagnostic improvements can be contributed via the normal contribution process; see [Contributing to Swift](https://www.swift.org/contributing/).
+The Swift evolution process covers all design changes, no matter how small, to the Swift language, its standard library, and the core tools necessary to build Swift programs.  This includes additions, removals, and changes to:
```

Comment:
should the be "Swift evolution process" or "Swift language evolution process"

cc @tkremenek @airspeedswift

Reply:
This document was written as if it just covered language evolution, but we've historically brought core tools proposals through the same evolution process.  I don't know of any way in which the process has been different for tools proposals, and I can't think of any reason we'd want it to be different.  At its core, the evolution process is just a sort of parliamentary procedure; the only time it touches on technical issues is the implementation requirement, and AFAIK we want to keep that for tools.  And I think it would be very confusing to community members if we had two subtly different processes, as well as creating all sorts of weird questions for proposals (like SE-0362) that cross the lines a bit.

As a result, I think the document needs to describe the evolution process as being common across the project, and this paragraph was my effort to define what's covered under it.  And although I think I've been clear about this, just to reiterate, this is not a power grab for the Language Workgroup; it's a common process, but different workgroups have specific granted authorities, and the Language Workgroup's authority does not extend to tools.  Tools evolution belongs to the Core Team unless/until the Core Team delegates that authority (as it did for language evolution).

Just for my own edification, I went through and tried to identify all of the tools proposals; almost all of them are about SwiftPM:
- [SE-0019](https://github.com/apple/swift-evolution/blob/main/proposals/0019-package-manager-testing.md)
- [SE-0038](https://github.com/apple/swift-evolution/blob/main/proposals/0038-swiftpm-c-language-targets.md)
- [SE-0063](https://github.com/apple/swift-evolution/blob/main/proposals/0063-swiftpm-system-module-search-paths.md)
- [SE-0082](https://github.com/apple/swift-evolution/blob/main/proposals/0082-swiftpm-package-edit.md)
- [SE-0085](https://github.com/apple/swift-evolution/blob/main/proposals/0085-package-manager-command-name.md)
- [SE-0129](https://github.com/apple/swift-evolution/blob/main/proposals/0129-package-manager-test-naming-conventions.md)
- [SE-0135](https://github.com/apple/swift-evolution/blob/main/proposals/0135-package-manager-support-for-differentiating-packages-by-swift-version.md)
- [SE-0145](https://github.com/apple/swift-evolution/blob/main/proposals/0145-package-manager-version-pinning.md)
- [SE-0146](https://github.com/apple/swift-evolution/blob/main/proposals/0146-package-manager-product-definitions.md)
- [SE-0149](https://github.com/apple/swift-evolution/blob/main/proposals/0149-package-manager-top-of-tree.md)
- [SE-0150](https://github.com/apple/swift-evolution/blob/main/proposals/0150-package-manager-branch-support.md)
- [SE-0151](https://github.com/apple/swift-evolution/blob/main/proposals/0151-package-manager-swift-language-compatibility-version.md)
- [SE-0152](https://github.com/apple/swift-evolution/blob/main/proposals/0152-package-manager-tools-version.md)
- [SE-0153](https://github.com/apple/swift-evolution/blob/main/proposals/0158-package-manager-manifest-api-redesign.md)
- [SE-0162](https://github.com/apple/swift-evolution/blob/main/proposals/0162-package-manager-custom-target-layouts.md)
- [SE-0175](https://github.com/apple/swift-evolution/blob/main/proposals/0175-package-manager-revised-dependency-resolution.md)
- [SE-0181](https://github.com/apple/swift-evolution/blob/main/proposals/0181-package-manager-cpp-language-version.md)
- [SE-0201](https://github.com/apple/swift-evolution/blob/main/proposals/0201-package-manager-local-dependencies.md)
- [SE-0208](https://github.com/apple/swift-evolution/blob/main/proposals/0208-package-manager-system-library-targets.md)
- [SE-0209](https://github.com/apple/swift-evolution/blob/main/proposals/0209-package-manager-swift-lang-version-update.md)
- [SE-0219](https://github.com/apple/swift-evolution/blob/main/proposals/0219-package-manager-dependency-mirroring.md)
- [SE-0226](https://github.com/apple/swift-evolution/blob/main/proposals/0226-package-manager-target-based-dep-resolution.md)
- [SE-0236](https://github.com/apple/swift-evolution/blob/main/proposals/0236-package-manager-platform-deployment-settings.md)
- [SE-0238](https://github.com/apple/swift-evolution/blob/main/proposals/0238-package-manager-build-settings.md)
- [SE-0250](https://github.com/apple/swift-evolution/blob/main/proposals/0238-package-manager-build-settings.md) (not SPM)
- [SE-0271](https://github.com/apple/swift-evolution/blob/main/proposals/0271-package-manager-resources.md)
- [SE-0272](https://github.com/apple/swift-evolution/blob/main/proposals/0272-swiftpm-binary-dependencies.md)
- [SE-0273](https://github.com/apple/swift-evolution/blob/main/proposals/0273-swiftpm-conditional-target-dependencies.md)
- [SE-0278](https://github.com/apple/swift-evolution/blob/main/proposals/0278-package-manager-localized-resources.md)
- [SE-0291](https://github.com/apple/swift-evolution/blob/main/proposals/0291-package-collections.md)
- [SE-0292](https://github.com/apple/swift-evolution/blob/main/proposals/0292-package-registry-service.md)
- [SE-0294](https://github.com/apple/swift-evolution/blob/main/proposals/0294-package-executable-targets.md)
- [SE-0301](https://github.com/apple/swift-evolution/blob/main/proposals/0301-package-editing-commands.md)
- [SE-0303](https://github.com/apple/swift-evolution/blob/main/proposals/0303-swiftpm-extensible-build-tools.md)
- [SE-0305](https://github.com/apple/swift-evolution/blob/main/proposals/0305-swiftpm-binary-target-improvements.md)
- [SE-0318](https://github.com/apple/swift-evolution/blob/main/proposals/0318-package-creation.md)
- [SE-0321](https://github.com/apple/swift-evolution/blob/main/proposals/0321-package-registry-publish.md)
- [SE-0325](https://github.com/apple/swift-evolution/blob/main/proposals/0325-swiftpm-additional-plugin-apis.md)
- [SE-0332](https://github.com/apple/swift-evolution/blob/main/proposals/0332-swiftpm-command-plugins.md)
- [SE-0339](https://github.com/apple/swift-evolution/blob/main/proposals/0339-module-aliasing-for-disambiguation.md)
- [SE-0342](https://github.com/apple/swift-evolution/blob/main/proposals/0342-static-link-runtime-libraries-by-default-on-supported-platforms.md)
- [SE-0356](https://github.com/apple/swift-evolution/blob/main/proposals/0356-swift-snippets.md)
- [SE-0362](https://github.com/apple/swift-evolution/blob/main/proposals/0362-piecemeal-future-features.md) (mostly a language feature, but with compiler/SPM aspects)

Reply:
Correct - most of them have been about SwiftPM. Based on that experience, I am of the opinion the while the evolution process we have in place is a great fit for language and API changes, it is less so for tools. With SwiftPM we had to wrangle with this problem in the past, where it is not obvious what changes require an evolution proposal and which do not. We also have a whole list of tools that are shipped as part of the toolchain and that do not follow the process at all / make changes as they see fit. As such, I suspect we end up with a slightly modified process for tools  that takes into account backwards compatibility but does not share some of the same API/ABI concerns the language and standard library process does

Reply:
I'm open to the idea that something about evolution ought to be different for tools.  However, I feel like those differences are basically something like using a different proposal template, which SwiftPM already does.  The actual evolution *process* is pretty simple and generic — write a proposal, pitch in public, do an open review, appointed experts make a decision — and I don't know what could be modified about it that wouldn't really be a major philosophical change.